src/sc/proofverifier.cpp

```diff
@@ -485,29 +485,28 @@ ProofVerificationResult CScProofVerifier::NormalVerifyCertificate(CCertProofVeri
     wrappedScVkeyPtr  sptrCertVk = input.verificationKey.GetVKeyPtr();
 
     bool ret = zendoo_verify_certificate_proof(
-        sptrConst.get(),
-        scidFe,
-        input.epochNumber,
-        input.quality,
-        bt_list_ptr,
-        bt_list_len,
-        custom_fields.get(),
-        custom_fields_len,
-        sptrCum.get(),
-        input.mainchainBackwardTransferRequestScFee,
-        input.forwardTransferScFee,
-        sptrProof.get(),
-        sptrCertVk.get(),
-        &code
-    );
-
-    if (!ret || code != CctpErrorCode::OK)
+               sptrConst.get(),
+               scidFe,
+               input.epochNumber,
+               input.quality,
+               bt_list_ptr,
+               bt_list_len,
+               custom_fields.get(),
+               custom_fields_len,
+               sptrCum.get(),
+               input.mainchainBackwardTransferRequestScFee,
+               input.forwardTransferScFee,
+               sptrProof.get(),
+               sptrCertVk.get(),
+               &code);
+
+    if (!ret)
     {
         LogPrintf("ERROR: %s():%d - certificate proof with ID [%d] does not verify: code [0x%x]\n",
-            __func__, __LINE__, input.proofId, code);
+                  __func__, __LINE__, input.proofId, code);
+        return ProofVerificationResult::Failed;
     }
-
-    return ret ? ProofVerificationResult::Passed : ProofVerificationResult::Failed;
+    return ProofVerificationResult::Passed;
```

Comment:
Given that it's never possible to end up in a state in which the verifier returned `true` and the error code is different from `CctpErrorCode::OK`, what about adding an assert?

```suggestion
    assert(code == CctpErrorCode::OK);
    return ProofVerificationResult::Passed;
```

In case, let's do the same for the function `NormalVerifyCsw()`.

Reply:
Makes sense, addressed in 597ad28380a26892ffce73359095e64c12dbdf0e.